soweego/validator/checks.py

```diff
@@ -520,6 +535,26 @@ def bio(
     return to_be_deprecated, _bio_to_be_added_generator(to_be_added), wd_bio
 
 
+def _apply_url_blacklist(url_statements):
+    LOGGER.info('Applying URL blacklist ...')
+    initial_input_size = len(url_statements)
+    blacklist = get_url_blacklist()
+
+    # O(nm) complexity: n = len(blacklist); m = len(url_statements)
```

Comment:
if this is a bottleneck you can make it faster via [tries](https://en.wikipedia.org/wiki/Trie) or [suffix trees](https://en.wikipedia.org/wiki/Suffix_tree). a trie is rather easy to implement but limited to queries like `a.startswith(b)`, while suffix trees are appropriate for full text search (pretty sure there's a library for that if you don't mind additional dependencies)

Reply:
Awesome idea, thanks a lot! While this is not a bottleneck, it's for sure an optimization worth to try: opening an issue for it.